File: order/domain/order.go
package domain

import "time"

// Order 订单聚合根
type Order struct {
	ID              int64
	TenantID        int64
	OrderNo         string
	BuyerID         int64
	BuyerHash       string // buyer_id + items hash，防重用
	Status          OrderStatus
	TotalAmount     int64 // 分
	DiscountAmount  int64
	FreightAmount   int64
	PayAmount       int64 // 实付
	RefundedAmount  int64 // 已退款
	CouponID        int64
	PaymentNo       string
	ReceiverName    string
	ReceiverPhone   string
	ReceiverAddress string
	Remark          string
	PaidAt          int64
	ShippedAt       int64
	ReceivedAt      int64
	ClosedAt        int64
	Items           []OrderItem
	Ctime           time.Time
	Utime           time.Time
}

type OrderStatus int32

const (
	OrderStatusPending   OrderStatus = 1
	OrderStatusPaid      OrderStatus = 2
	OrderStatusShipped   OrderStatus = 3
	OrderStatusReceived  OrderStatus = 4
	OrderStatusCompleted OrderStatus = 5
	OrderStatusCancelled OrderStatus = 6
	OrderStatusRefunding OrderStatus = 7
	OrderStatusRefunded  OrderStatus = 8
)

type OrderItem struct {
	ID               int64
	OrderID          int64
	TenantID         int64
	ProductID        int64
	SKUID            int64
	ProductName      string
	SKUSpec          string
	Image            string
	Price            int64 // 分
	Quantity         int32
	Subtotal         int64 // price * quantity
	RefundedQuantity int32
	Ctime            time.Time
}

type OrderStatusLog struct {
	ID           int64
	OrderID      int64
	FromStatus   OrderStatus
	ToStatus     OrderStatus
	OperatorID   int64
	OperatorType int32 // 1=买家 2=商家 3=平台 4=系统
	Remark       string
	Ctime        time.Time
}

type RefundOrder struct {
	ID           int64
	TenantID     int64
	OrderID      int64
	RefundNo     string
	BuyerID      int64
	Type         int32 // 1=仅退款 2=退货退款
	Status       RefundStatus
	RefundAmount int64
	Reason       string
	RejectReason string
	Items        string // JSON: [{"sku_id":1,"quantity":2}]
	Ctime        time.Time
	Utime        time.Time
}

type RefundStatus int32

const (
	RefundStatusPending   RefundStatus = 1 // 待审核
	RefundStatusApproved  RefundStatus = 2 // 审核通过
	RefundStatusRefunding RefundStatus = 3 // 退款中
	RefundStatusRefunded  RefundStatus = 4 // 已退款
	RefundStatusRejected  RefundStatus = 5 // 已拒绝
)

File: frontend/src/api/order.ts
import { request } from './client'

export interface CreateOrderParams {
  items: Array<{ skuId: number; quantity: number }>
  addressId: number
  couponId?: number
  remark?: string
}

export interface CreateOrderResult {
  orderNo: string
  payAmount: number
}

export interface OrderItem {
  id: number
  orderId: number
  productId: number
  skuId: number
  productName: string
  skuSpec: string
  productImage: string
  price: number
  quantity: number
  totalAmount: number
}

export interface Order {
  id: number
  orderNo: string
  status: number
  totalAmount: number
  discountAmount: number
  freightAmount: number
  payAmount: number
  receiverName: string
  receiverPhone: string
  receiverAddress: string
  remark: string
  payTime: number
  shipTime: number
  receiveTime: number
  closeTime: number
  items: OrderItem[]
  ctime: string
  utime: string
}

// status: 1=待付款, 2=已付款/待发货, 3=已发货, 4=已收货, 5=已完成, 6=已取消, 7=已退款

export interface RefundOrder {
  id: number
  orderId: number
  refundNo: string
  type: number
  status: number
  refundAmount: number
  reason: string
  ctime: string
  utime: string
}

// refund status: 1=待审核, 2=已通过, 3=退款中, 4=已完成, 5=已拒绝

export interface PageResult<T> {
  list: T[]
  total: number
}

export function createOrder(params: CreateOrderParams) {
  return request<CreateOrderResult>({ method: 'POST', url: '/orders', data: params })
}

export function listOrders(params: { status?: number; page: number; pageSize: number }) {
  return request<PageResult<Order>>({ method: 'GET', url: '/orders', params })
}

export function getOrder(orderNo: string) {
  return request<Order>({ method: 'GET', url: `/orders/${orderNo}` })
}

export function cancelOrder(orderNo: string) {
  return request<void>({ method: 'POST', url: `/orders/${orderNo}/cancel` })
}

export function confirmReceive(orderNo: string) {
  return request<void>({ method: 'POST', url: `/orders/${orderNo}/confirm` })
}

export function applyRefund(orderNo: string, params: { reason: string }) {
  return request<{ refundNo: string }>({ method: 'POST', url: `/orders/${orderNo}/refund`, data: params })
}

export function listRefunds(params: { page: number; pageSize: number }) {
  return request<PageResult<RefundOrder>>({ method: 'GET', url: '/refunds', params })
}

export function getRefund(refundNo: string) {
  return request<RefundOrder>({ method: 'GET', url: `/refunds/${refundNo}` })
}
